mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainController.java

```diff
@@ -139,69 +135,115 @@
             LOGGER.error("{} service domain cluster not found", sdcName);
-            setStatusCondition(sd, CONDITION_READY, Boolean.FALSE);
-            setStatusCondition(sd, CONDITION_SERVICE_DOMAIN_CLUSTER_READY, REASON_SDC, sdcName + " " + MESSAGE_SDC_NOT_FOUND, Boolean.FALSE);
-            return updateStatus(sd);
+            return updateStatusWithCondition(sd, new ConditionBuilder()
+                    .withType(CONDITION_SERVICE_DOMAIN_CLUSTER_READY)
+                    .withStatus(STATUS_FALSE)
+                    .withReason(REASON_SDC)
+                    .withMessage(sdcName + " " + MESSAGE_SDC_NOT_FOUND)
+                    .build());
         }
 
-        if (sdc.getStatus().getCondition(CONDITION_READY) == null || Boolean.FALSE.toString().equalsIgnoreCase(sdc.getStatus().getCondition(CONDITION_READY).getStatus())) {
+        if (sdc.getStatus() != null && sdc.getStatus().getCondition(CONDITION_READY) == null || Boolean.FALSE.toString().equalsIgnoreCase(sdc.getStatus().getCondition(CONDITION_READY).getStatus())) {
             LOGGER.error("{} service domain cluster not ready", sdcName);
-            setStatusCondition(sd, CONDITION_READY, Boolean.FALSE);
-            setStatusCondition(sd, CONDITION_SERVICE_DOMAIN_CLUSTER_READY, REASON_SDC, sdcName + " " + MESSAGE_SDC_NOT_READY, Boolean.FALSE);
-            return updateStatus(sd);
+            return updateStatusWithCondition(sd, new ConditionBuilder()
+                    .withType(CONDITION_SERVICE_DOMAIN_CLUSTER_READY)
+                    .withStatus(STATUS_FALSE)
+                    .withReason(REASON_SDC)
+                    .withMessage(sdcName + " " + MESSAGE_SDC_NOT_READY)
+                    .build());
         }
 
-        setStatusCondition(sd, CONDITION_SERVICE_DOMAIN_CLUSTER_READY, Boolean.TRUE);
+        control = updateStatusWithReadyCondition(sd, CONDITION_SERVICE_DOMAIN_CLUSTER_READY);
+        if(control.isUpdateStatus()){
+            return control;
+        }
 
         try {
             createOrUpdateDeployment(sd, sdc.getStatus().getKafkaBroker());
             createOrUpdateService(sd);
             if (sd.getSpec().getExpose() != null && sd.getSpec().getExpose().contains(ServiceDomainSpec.ExposeType.http)) {
-                setStatusCondition(sd, CONDITION_INTEGRATION_READY, REASON_INTEGRATION, MESSAGE_INTEGRATION_NOT_READY, Boolean.FALSE);
                 final String sdConfigMapName = "integration-" + toLowerHyphen(sd.getSpec().getType().toString()) + "-http";
                 ConfigMap sdConfigMap = client.configMaps().inNamespace(client.getNamespace()).withName(sdConfigMapName).get();
+                final ServiceDomainSpec.Type sdType = sd.getSpec().getType();
+                final String sdTypeAsString = toLowerHyphen(sdType.toString());
 
-                String validationError = validateSdConfigMap(sd, sdConfigMapName, sdConfigMap);
-                if (validationError != null) {
+                if (sdConfigMap == null) {
```

Comment:
All the integration checks should be part of the createOrUpdateIntegration, create submethods if necessary